File: gameserver/src/main/java/l2s/gameserver/network/l2/c2s/RequestPCCafeCouponUse.java
package l2s.gameserver.network.l2.c2s;

/**
 * format: chS
 */
public class RequestPCCafeCouponUse implements IClientIncomingPacket
{
	// format: (ch)S
	private String _unknown;

	@Override
	public boolean readImpl(l2s.gameserver.network.l2.GameClient client, l2s.commons.network.PacketReader packet)
	{
		_unknown = packet.readS();
		return true;
	}

	@Override
	public void run(l2s.gameserver.network.l2.GameClient client)
	{
		//TODO not implemented
	}
}

File: gameserver/src/main/java/l2s/gameserver/network/l2/c2s/coupon/RequestPCCafeCouponUse.kt
package l2s.gameserver.network.l2.c2s.coupon

import l2s.gameserver.network.l2.c2s.L2GameClientPacket

class RequestPCCafeCouponUse : L2GameClientPacket() {

    private lateinit var enteredCode: String

    override fun readImpl() {
        enteredCode = readS()
    }

    override fun runImpl() {}

}
